tuta-sdk/rust/src/entity_client.rs

```diff
@@ -0,0 +1,180 @@
+use std::sync::Arc;
+
+use crate::{ApiCallError, AuthHeadersProvider, IdTuple, RestClient, TypeRef};
+use crate::element_value::{ElementValue, ParsedEntity};
+use crate::instance_mapper::InstanceMapper;
+use crate::json_element::RawEntity;
+use crate::rest_client::{HttpMethod, RestClientOptions};
+use crate::rest_error::{NetworkError};
+
+/// The value used to identify an entity/instance within the backend
+pub enum IdType {
+    Single(String),
+    Tuple(IdTuple),
+}
+
+/// A high level interface to manipulate entities/instances via the REST API
+pub struct EntityClient {
+    rest_client: Arc<dyn RestClient>,
+    base_url: String,
+    auth_headers_provider: Arc<dyn AuthHeadersProvider + Send + Sync>,
+    instance_mapper: Arc<InstanceMapper>,
+}
+
+impl EntityClient {
+    pub(crate) fn new(
+        rest_client: Arc<dyn RestClient>,
+        instance_mapper: Arc<InstanceMapper>,
+        base_url: &str,
+        auth_headers_provider: Arc<dyn AuthHeadersProvider + Send + Sync>,
+    ) -> Self {
+        EntityClient {
+            rest_client,
+            instance_mapper,
+            base_url: base_url.to_owned(),
+            auth_headers_provider,
+        }
+    }
+}
+
+impl EntityClient {
+    /// Gets a LET (List Element Type) entity/instance of type `type_ref` from the backend
+    pub async fn load_list_element(
```

Comment:
we should either have a single `load()` function or only handle one type of ID in method. My initial idea was to have `load_element` to load element types (with single id) or list element types (with IdTuple). Currently our ts code has one `load()` method. I don't mind which option we chose but we shouldn't have both that do the same thing

Reply:
We (me & wec) decided to merge the two in 43637b691625ae7959c0b119264550a48d5d6c94.